pkg/kubelet/images/image_manager.go

```diff
@@ -155,6 +205,25 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 	}
 	m.logIt(ref, v1.EventTypeNormal, events.PulledImage, logPrefix, fmt.Sprintf("Successfully pulled image %q in %v", container.Image, time.Since(startTime)), klog.Info)
 	m.backOff.GC()
+
+	m.lock.Lock()
+	if imagePullResult.pullCredentialsHash == "" {
+		// successful pull no auth hash returned, auth was not required so we should reset the hashmap for this
+		// imageref since auth is no longer required for the local image cache, allowing use of the ImageRef
+		// by other pods if it remains cached and pull policy is PullIfNotPresent
+		delete(m.ensureSecretPulledImage, imageRef)
+	} else {
+		// store/create hashMatch map entry for auth config hash key used to pull the image
+		// for this imageref (digest)
+		digest := m.ensureSecretPulledImage[imageRef]
```

Comment:
better to add the `ok` like: `digest, ok := m.ensureSecretPulledImage[imageRef]`

Reply:
Agree.